src/sku/molecule/domain/src/entities/molecule_data_room_activity.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::{DateTime, Utc};
+use file_utils::MediaType;
+use internal_error::{InternalError, ResultIntoInternal};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Clone, Copy, Debug, serde::Deserialize, serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+pub enum MoleculeDataRoomFileActivityType {
+    Added,
+    Updated,
+    Removed,
+}
+
+// TODO: revisit after IPNFT-less projects changes.
+#[derive(Debug)]
+pub struct MoleculeDataRoomActivityEntity {
+    pub system_time: DateTime<Utc>,
+
+    pub event_time: DateTime<Utc>,
+
+    pub activity_type: MoleculeDataRoomFileActivityType,
+
+    pub ipnft_uid: String,
```

Comment:
Outside of this particular portion, let's think of a higher level type within Molecule entities for these IDs.